### ISSUE TITLE
Integer divison `a // b` is broken with NVFuser. Broken decomposition and/or meta?

### ISSUE THREAD

**Issue by nikitaved** [Lightning-AI/lightning-thunder#808]
Take a look at the following repro:
```python
In [1]: import torch, thunder

In [2]: def f(x, y):
   ...:     return x // y
   ...: 

In [3]: x = torch.randint(1, 10, (3, 3), device="cuda")

In [4]: y = torch.randint(1, 10, (3, 3), device="cuda")

In [5]: f(x, y)
Out[5]: 
tensor([[0, 0, 0],
        [0, 1, 0],
        [0, 1, 4]], device='cuda:0')

In [6]: jf = thunder.jit(f)

In [7]: jf(x, y)
Out[7]: 
tensor([[0.8889, 0.7500, 0.5000],
        [0.4444, 1.0000, 0.4286],
        [0.3333, 1.5000, 4.0000]], device='cuda:0')

In [8]: thunder.last_traces(jf)[-1]
Out[8]: 
# Constructed by Delete Last Used (took 0 milliseconds)
import torch
from thunder.executors.torchex import no_autocast

@torch.no_grad()
@no_autocast
def computation(x, y):
  # x: "cuda:0 i64[3, 3]"
  # y: "cuda:0 i64[3, 3]"
  [result] = nvFusion0(x, y)
    # t0 = prims.signbit(x)  # t0: "cuda:0 b8[3, 3]"
    # t1 = prims.signbit(y)  # t1: "cuda:0 b8[3, 3]"
    # t2 = prims.ne(t0, t1)  # t2: "cuda:0 b8[3, 3]"
    # t3 = prims.fmod(x, y)  # t3: "cuda:0 i64[3, 3]"
    # t4 = prims.ne(t3, 0)  # t4: "cuda:0 b8[3, 3]"
    # t5 = prims.bitwise_and(t2, t4)  # t5: "cuda:0 b8[3, 3]"
    # t6 = prims.div(x, y)  # t6: "cuda:0 i64[3, 3]"
    # t7 = prims.convert_element_type(t5, dtypes.int64)  # t7: "cuda:0 i64[3, 3]"
    # result = prims.sub(t6, t7)  # result: "cuda:0 i64[3, 3]"
  del x, y
  return result
```
We definitely have a gap between the true result and the meta. We should also inspect the decomposition itself, because I am not sure that `prims.sub(t6, t7)` makes sense, since `t7` is the result of `bitwise_and` of boolean masks and `t6 = prims.div(x, y)`...

**Comment by t-vi** [comment]
I think what it's trying to achieve is to subtract one from the result when both inputs are negative because we have a misalignment between the PyTorch // and our primitive `div`.

```
In [7]: a
Out[7]: tensor([-5, -4, -3, -2, -1,  0,  1,  2,  3,  4])

In [8]: b
Out[8]: 
tensor([[-2],
        [-1],
        [ 1],
        [ 2]])

In [9]: a // b
Out[9]: 
tensor([[ 2,  2,  1,  1,  0,  0, -1, -1, -2, -2],
        [ 5,  4,  3,  2,  1,  0, -1, -2, -3, -4],
        [-5, -4, -3, -2, -1,  0,  1,  2,  3,  4],
        [-3, -2, -2, -1, -1,  0,  0,  1,  1,  2]])

In [10]: 
```
